### PR TITLE
Clean-up SupportsParamCoercion overrides from functions

### PR DESCRIPTION
**Pull request by CarlosFigueiraMSFT**
The default value is true for the functions, but many functions override this property with the same value anyway. This PR removes such overrides, and also some other overrides which are not necessary.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Coalesce.cs [hunk 1]
```diff
@@ -22,8 +22,6 @@ internal sealed class CoalesceFunction : BuiltinFunction
     {
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Coalesce.cs:25]: >SupportsParamCoercion => false; [](http://example.com/codeflow?start=29&length=31)

Coercion is done in CheckTypes if the function takes arguments of different types

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Count.cs [hunk 2]
```diff
@@ -22,8 +22,6 @@ internal sealed class CountFunction : FunctionWithTableInput
     {
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Count.cs:25]: >SupportsParamCoercion [](http://example.com/codeflow?start=29&length=21)

The base class (FunctionWithTableInput) already forbids coercions for the first argument (and this function only has one)

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountA.cs [hunk 3]
```diff
@@ -22,8 +22,6 @@ internal sealed class CountAFunction : FunctionWithTableInput
     {
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountA.cs:25]: >SupportsParamCoercion => false [](http://example.com/codeflow?start=29&length=30)

The base class (FunctionWithTableInput) already forbids coercions for the first argument (and this function only has one)

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountRows.cs [hunk 4]
```diff
@@ -24,8 +24,6 @@ internal sealed class CountRowsFunction : FunctionWithTableInput
 
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/CountRows.cs:27]: >SupportsParamCoercion => false [](http://example.com/codeflow?start=29&length=30)

The base class (FunctionWithTableInput) already forbids coercions for the first argument (and this function only has one)

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FirstLast.cs [hunk 5]
```diff
@@ -24,8 +24,6 @@ internal sealed class FirstLastFunction : FunctionWithTableInput
     {
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FirstLast.cs:27]: >SupportsParamCoercion => false [](http://example.com/codeflow?start=29&length=30)

The base class (FunctionWithTableInput) already forbids coercions for the first argument (and this function only has one)

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsError.cs [hunk 6]
```diff
@@ -17,8 +17,6 @@ internal sealed class IsErrorFunction : BuiltinFunction
     {
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsError.cs:20]: >SupportsParamCoercion => false [](http://example.com/codeflow?start=29&length=30)

This function can take any argument type, there's no coercion anyway

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsNumeric.cs [hunk 7]
```diff
@@ -13,8 +13,6 @@ internal sealed class IsNumericFunction : BuiltinFunction
     {
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsNumeric.cs:16]: >SupportsParamCoercion => false [](http://example.com/codeflow?start=29&length=30)

This function can take any argument type already

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Shuffle.cs [hunk 8]
```diff
@@ -12,15 +12,13 @@
...
     // Shuffle(source:*)
-    internal sealed class ShuffleFunction : BuiltinFunction
+    internal sealed class ShuffleFunction : FunctionWithTableInput
     {
         // Multiple invocations with the same args may result in different results.
         public override bool IsStateless => false;
 
         public override bool IsSelfContained => true;
 
-        public override bool SupportsParamCoercion => false;
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Shuffle.cs:22]: >SupportsParamCoercion => false [](http://example.com/codeflow?start=29&length=30)

The (new) base class (FunctionWithTableInput) already forbids coercions for the first argument (and this function only has one)

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**